src/bundles/gateway.js

```diff
@@ -51,6 +51,109 @@ export const checkValidHttpUrl = (value) => {
   return url.protocol === 'http:' || url.protocol === 'https:'
 }
 
+/**
+ * Security validation for subdomain gateway URLs
+ * @param {string} gatewayUrl - The gateway URL to validate
+ * @returns {{isValid: boolean, warnings: string[], errors: string[], securityScore: number}} - Validation result with security warnings
+ */
+export const validateSubdomainGatewaySecurity = (gatewayUrl) => {
+  const warnings = []
+  const errors = []
+
+  try {
+    const url = new URL(gatewayUrl)
+
+    // Check for HTTPS
+    if (url.protocol !== 'https:') {
+      warnings.push('Gateway should use HTTPS for security')
+    }
+
+    // Check for localhost/private IPs in production
+    const hostname = url.hostname.toLowerCase()
+    const isLocalhost = hostname === 'localhost' ||
+                       hostname === '127.0.0.1' ||
+                       hostname.startsWith('192.168.') ||
+                       hostname.startsWith('10.') ||
+                       hostname.startsWith('172.')
+
+    if (isLocalhost) {
+      warnings.push('Using localhost/private IP gateway may not be accessible to other users')
+    }
+
+    // Check for potentially risky subdomain patterns
+    if (hostname.includes('ipfs.') && !hostname.startsWith('ipfs.')) {
+      warnings.push('Subdomain gateway with "ipfs." in the middle may cause confusion')
+    }
+
+    // Check for suspicious domains
+    const suspiciousPatterns = [
+      /\.tk$/, /\.ml$/, /\.ga$/, /\.cf$/, // Free domains often used for malicious purposes
+      /\.onion$/, // Tor hidden services
+      /localhost\./, /127\.0\.0\.1\./, /0\.0\.0\.0\./ // Localhost subdomains
+    ]
+
+    for (const pattern of suspiciousPatterns) {
+      if (pattern.test(hostname)) {
+        warnings.push('Gateway domain may be unreliable or potentially malicious')
+        break
+      }
+    }
```

Comment:
This is not a place to make judgement about domain being safe or not.

We should only test if domain is a valid subdomain gateway and if it correctly implements URL convention and CORS headers. 

Please remove this.